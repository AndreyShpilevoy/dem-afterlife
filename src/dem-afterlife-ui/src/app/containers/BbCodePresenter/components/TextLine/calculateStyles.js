const calculateStyles = ({themeName}) => ({
    text: {
        whiteSpace: 'pre-wrap'
    },
    options: {meta: 'TextLine', themeName}
});

export default calculateStyles;