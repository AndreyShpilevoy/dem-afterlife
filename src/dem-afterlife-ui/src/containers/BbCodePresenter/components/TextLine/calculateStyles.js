const calculateStyles = () => ({
  text: {
    whiteSpace: "pre-wrap"
  }
});

export default calculateStyles;
