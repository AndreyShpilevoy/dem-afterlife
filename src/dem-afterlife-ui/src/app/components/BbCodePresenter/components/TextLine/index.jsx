import React from 'react';
import {node, shape} from 'prop-types';
import {withStyles} from 'styles';
import calculateStyles from './calculateStyles';

const TextLine = ({children, styles}) => (
    <span
        className={styles.text}>
        {children}
    </span>
);

TextLine.propTypes = {
    styles: shape().isRequired,
    children: node.isRequired
};

export default withStyles(theme => calculateStyles(theme))(TextLine);
