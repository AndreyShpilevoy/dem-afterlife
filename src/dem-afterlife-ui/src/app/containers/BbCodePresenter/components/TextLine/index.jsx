import React from 'react';
import {node, shape} from 'prop-types';
import {withStyles} from 'styles';
import calculateStyles from './calculateStyles';

export const TextLinePure = ({children, styles}) => (
    <span
        className={styles.text}>
        {children}
    </span>
);

TextLinePure.propTypes = {
    styles: shape().isRequired,
    children: node.isRequired
};

export default withStyles(theme => calculateStyles(theme))(TextLinePure);
