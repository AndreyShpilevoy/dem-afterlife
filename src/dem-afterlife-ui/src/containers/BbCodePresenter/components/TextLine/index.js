import React from "react";
import { node, shape } from "prop-types";
import { injectSheet } from "styles";
import calculateStyles from "./calculateStyles";

export const TextLinePure = ({ children, classes }) => (
  <span className={classes.text}>{children}</span>
);

TextLinePure.propTypes = {
  classes: shape().isRequired,
  children: node.isRequired
};

export default injectSheet(calculateStyles, { componentName: "TextLine" })(
  TextLinePure
);
