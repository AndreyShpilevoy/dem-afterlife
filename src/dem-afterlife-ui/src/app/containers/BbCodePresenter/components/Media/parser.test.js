/* eslint-disable react/jsx-filename-extension, no-undef, fp/no-unused-expression, fp/no-nil, max-statements */

import React from 'react';
import TextLine from '../TextLine';
import parseTextLineComponentToEmbedLink from './parser';

const firstParser = jest.fn(string => string === 'valid' ?
    {
        type: 'iframe',
        success: true,
        url: 'https://Valid',
        shortHeight: false
    } :
    {success: false});

const secondParser = jest.fn(() => ({success: false}));
const thirdParser = jest.fn(() => ({success: false}));

describe('Media parseTextLineComponentToEmbedLink', () => {
    it('should return default result if sent wrong first argument', () => {
        expect(parseTextLineComponentToEmbedLink('TextLine', [firstParser])).toMatchSnapshot();
    });

    it('should return default result if sent empty array as second argument', () => {
        expect(parseTextLineComponentToEmbedLink(<TextLine>{'valid'}</TextLine>, [])).toMatchSnapshot();
    });

    it('should return default result if sent correct arguments, but with wrong data', () => {
        expect(parseTextLineComponentToEmbedLink(<TextLine>{'some wrong content'}</TextLine>, [firstParser])).toMatchSnapshot();
    });

    it('should return handled result if data was correct', () => {
        expect(parseTextLineComponentToEmbedLink(<TextLine>{'valid'}</TextLine>,
            [secondParser, firstParser, thirdParser])).toMatchSnapshot();
    });
});
