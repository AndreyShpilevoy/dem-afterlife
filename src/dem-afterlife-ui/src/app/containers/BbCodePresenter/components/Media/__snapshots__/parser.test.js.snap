// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media parseTextLineComponentToEmbedLink should return default result if sent correct arguments, but with wrong data 1`] = `
Object {
  "success": false,
  "type": "none",
  "url": "",
}
`;

exports[`Media parseTextLineComponentToEmbedLink should return default result if sent empty array as second argument 1`] = `
Object {
  "success": false,
  "type": "none",
  "url": "",
}
`;

exports[`Media parseTextLineComponentToEmbedLink should return default result if sent wrong first argument 1`] = `
Object {
  "success": false,
  "type": "none",
  "url": "",
}
`;

exports[`Media parseTextLineComponentToEmbedLink should return handled result if data was correct 1`] = `
Object {
  "shortHeight": false,
  "success": true,
  "type": "iframe",
  "url": "https://Valid",
}
`;
